### PR TITLE
Wip ui redesign properties

### PR DESCRIPTION
**Pull request by mjourdan**
Hi! Here are some more changes for the properties pane. LabelEditor is enabled (though editting existing labels produces weird behavior) and some visual changes happened. Regards

### CODE REVIEW

File: src/paperwork/frontend/mainwindow/__init__.py [hunk 1]
```diff
@@ -2474,13 +2485,15 @@ def _clear_label_list(self):
     def _readd_label_widgets(self, labels):
         label_widgets = {}
         self.widgets['labels'].freeze_child_notify()
+        #self.widgets['labels'].set_border_width(20)
```
Comment (jflesch) [inline src/paperwork/frontend/mainwindow/__init__.py]: Please do not let dead code going around like that.
Either there is a good reason to keep it, in which case you should add a comment explaining why, or there isn't.

File: src/paperwork/frontend/util/__init__.py [hunk 2]
```diff
@@ -69,7 +69,7 @@ def load_uifile(filename):
 
 def load_cssfile(filename):
     """
-    Load a .glade file and return the corresponding widget tree
```
Comment (jflesch) [inline src/paperwork/frontend/util/__init__.py]: Woops, nice catch :)

**Comment by jflesch** [comment]
It looks awesome :). I'm going to merge it as is.

I just have one other remark. When I open the label editor dialog, I get the following warnings :
```
INFO   root                           UI file used: src/paperwork/frontend/labeleditor/labeleditor.glade

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkColorPlane': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkGrid': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkOverlay': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkColorEditor': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkBox': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkColorChooserWidget': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkVBox': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkBox': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkDialog': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkColorPlane': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkGrid': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkOverlay': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkColorEditor': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkBox': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkColorChooserWidget': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkVBox': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkBox': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkDialog': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkColorPlane': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkGrid': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkOverlay': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkColorEditor': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkBox': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkColorChooserWidget': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkVBox': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkBox': NULL pointer

(paperwork:10026): Gtk-WARNING **: drawing failure for widget `GtkDialog': NULL pointer
```

Also, the label editor dialog looks a little bit weird/empty until we start selecting a color. I haven't had much time to look at it, but wouldn't be possible to preselect a color ?

**Comment by jflesch** [comment]
Just to clarify, here is what I see:
* Before : http://jflesch.kwain.net/~jflesch/paperwork/no_color_selected_yet.png
* After : http://jflesch.kwain.net/~jflesch/paperwork/color_selected.png

**Comment by jflesch** [comment]
Hm, also, label edition has 2 bugs:
- the color is not correctly pre-selected in the label editor dialog
- closing the dialog is always considered as a cancelation ("WARNING paperwork.frontend.mainwindow  Label edition cancelled")

**Comment by jflesch** [comment]
Anyway, again, great work ! :)

**Comment by mjourdan** [comment]
> Please do not let dead code going around like that.

I couldn't figure out why I couldn't get no border for the listbox, and let that awful line... My bad! 

> I just have one other remark. When I open the label editor dialog, I get the following warnings [...]

Yeah, I noticed that too. I thought this was some bug you were aware of, as it appeared before I made any change to the label editor. Anyway, to me the widget is too complex. Unless there is a real need of transparency, I would like to see the color picker cleaned of its alpha channel, to look more like [this](https://raw.githubusercontent.com/mjourdan/gnome-design/master/paperwork/wires/first-paper.png) (near the bottom of the page).

> Also, the label editor dialog looks a little bit weird/empty until we start selecting a color. I haven't had much time to look at it, but wouldn't it be possible to preselect a color ?

Indeed. It would certainly be nice to preselect a (random but different enough from other labels) color.

> - the color is not correctly pre-selected in the label editor dialog

This is a really weird one. I couldn't figure out how the label text could come from the selected row, while the color could come from the previous selection. I wonder if it has to do with some kind of simultaneous changes happening in ActionEditLabel and on_button_clicked.

> - closing the dialog is always considered as a cancelation ("WARNING paperwork.frontend.mainwindow Label edition cancelled")

Didn't notice this one. I was successful in both creating new labels and editing existing labels, though.

> Anyway, again, great work ! :)

Glad you like it!

**Comment by jflesch** [comment]
> I thought this was some bug you were aware of, as it appeared before I made any change

Hm, I didn't notice it before :/

**Comment by mjourdan** [comment]
> Hm, I didn't notice it before :/. I will see if I have time to have a look.

Removing the GtkColorChooserWidget from the .glade file makes the issue gone. I don't know if this color stuff make any use of cairo, but [that could be related to it](http://www.gtkforums.com/viewtopic.php?f=3&t=178724).

**Comment by mjourdan** [comment]
I found that all our troubles are gone if 'show_editor' is set to False. This seems to solve our 3 issues:
- gtk warnings
- the label editor dialog looks weird (headerbar is black, the color square is invisible)
- the color is not correctly pre-selected in the label editor dialog

If we use the "+" button of the palette, it opens the editor, which then works just fine. So, you'll find in a future pull request a single-line commit for this.

**Pull request merged**